### DISCUSSION TITLE
Type declaration in reduce breaking formatting?

### DISCUSSION THREAD

**Discussion by RashiqAzhan** [prettier/prettier#16328] [Q&A] [answered]
Is this expected behavior? It's more than 80 lines wide.
[Playground Link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEANtAOYBKcAJumHAKIAeADgE5wDO7eUAFPAzCT52MFjigkAlEJFiJAbQC6+YAB0o+TfjYx0LDfNUgAhkYA0+IwCMji9Vq0A6ALbGmPHmEn4AvAD57By0wQKCndiZCHBgeAHpVVXYAalipULD8RzZqWgAedIzhUXESJQKgvw92CGc4ABlSADljWoswPTZYBokANWNCdDgASSgAYW9-fB5q2u6SZtrHYxiABkkAfkdCBBIYAAt8dfX8Ffw-fHaWTpg5voHhsa2d-cPhGvqmlrh8IXlL69u-UGI1GikkFiU4PKknKjlQhGWPEkAG5AgBfEBmEAQJi4aDsZCgYxXCAAdwACsSEASUP1ScYAJ4ErFWFjGMAAazgMAAyl86uI4MhUP12HAWWzOdyeUx2SVkKJBlixc4cAqWEqQHBmHAxLVYP0ACq6qDEnAcYWi8UgLgSbYARXQEHglsIYqxACt2AweSUHU6XUgRW7rQBHANwcksHE0kzsAC0UDgVComJAomMOCiElGNVcyBMhEIadtJG2AEEYMUrOh4OTdQKk673SA9jBnIQAOp7aIcWW0HnU6I4ABu0QZBbAnDTI+BUEoCF5YDEuPL855MAZ22b1tYEDFnbZTALrA4upHQqx4jFLBgUeMJHzQatWNlLBvBc3TA4y5wuLTrDiDAnY4JQ+zIAAHCsWJsOGOBsPej7GDuWIwMYVggWBezIAATFi6Bioa6E0sGLZwM4VhUAulB1MYEjoA+cAAGIQCwrhVvKtK1hAIBomiQA)

```ts
function longReduceExpression(text: string): string[] {
    return ["a", "b"]
      .map((c) =>
        c
          .split(/\s+/g)
          .reduce<
            string[]
          >((someLongName, currentLongValueInC) => (someLongName.at(0)?.length ?? 0 > currentLongValueInC.length ? someLongName : [currentLongValueInC]), []),
      )
      .flat();
  }
```

![image](https://github.com/prettier/prettier/assets/7370540/7b627ca6-a324-4926-a14e-53f61a111ed7)

Works fine if the `reduce<string[]>(...` is removed.

![image](https://github.com/prettier/prettier/assets/7370540/3351784c-5a4f-42c1-8831-61fc56adde34)

Prettier version 3.2.5.
Formatted using `npx prettier --write .`

**Comment by kachkaev** [comment, marked as answer]
👋 @RashiqAzhan! I guess that this bug is tracked in https://github.com/prettier/prettier/issues/15918.
